akka-projection-persistence-query/src/main/scala/akka/persistence/query/TypedEventEnvelope.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.query
+
+object TypedEventEnvelope {
```

Comment:
And rename back to `TypedEventEnvelope` if we move it to Akka?

Reply:
If we can't add the type to the existing one in a compatible way I don't think it's worth moving to Akka. Too much changes everywhere (in all plugins).

Reply:
I was thinking on something as an adapter to avoid change the existing API, but if we can add it to the existing one in a compatible way, that's event better.